# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/PendingStyles.h
# Commit: 5bacd687fa98
# Full Hash: 5bacd687fa98b0e9a5dea9944f360fdc5a6770e6
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2023-01-17 16:13:02
# Regressor Bug: 1807829
# File Overlap Count: 2
# Description:
#   Bug 1807829 - Make `HTMLEditor::CreateStyleForInsertText` use same logic as `HTMLEditor::SetInlinePropertiesAsSubAction` to apply new style r=m_kato
#   
#   The problem in Yahoo mail is, when you apply new font size to:
#   ```
#   <div><font size="7">{}<br></font></div>
# ==============================================================================

diff -r a7465d2d89ba -r 5bacd687fa98 editor/libeditor/PendingStyles.h
--- a/editor/libeditor/PendingStyles.h	Mon Jan 16 23:14:12 2023 +0000
+++ b/editor/libeditor/PendingStyles.h	Tue Jan 17 05:26:14 2023 +0000
@@ -263,6 +263,13 @@
   }
 
   /**
+   * TakeAllPreservedStyles() moves all preserved styles and values to
+   * aOutStylesAndValues.
+   */
+  void TakeAllPreservedStyles(
+      nsTArray<EditorInlineStyleAndValue>& aOutStylesAndValues);
+
+  /**
    * TakeRelativeFontSize() hands back relative font value, which is then
    * cleared out.
    */